### PR TITLE
SAK-500 narrowing changelog parameters

### PR DESCRIPTION
**Pull request by rafalkanski**
@filipgolonka @bartosz-kahl

### CODE REVIEW

File: spec/Snt/Capi/Repository/Article/FindParametersSpec.php [hunk 1]
```diff
@@ -39,6 +41,21 @@ function it_creates_find_parameters_for_publication_id()
...
 
+    function it_creates_find_parameters_for_publication_id_with_time_range_and_limit()
+    {
+        $timeRange = new TimeRange(
+            new DateTime('2016-01-01'),
+            new DateTime('2016-02-01')
+        );
+        $limit = 2;
+
+        $this->beConstructedThrough('createForPublicationIdWithTimeRangeAndLimit',[PublicationId::AP, $timeRange, $limit]);
```
Comment (filipgolonka) [inline spec/Snt/Capi/Repository/Article/FindParametersSpec.php]: incorrect formatting, missing one space after `,`

File: spec/Snt/Capi/Repository/Article/FindParametersSpec.php [hunk 2]
```diff
@@ -73,4 +90,17 @@ function it_builds_empty_string_when_only_publication_id_is_present()
...
+
+    function it_builds_query()
+    {
+        $timeRange = new TimeRange(
+            new DateTime('2016-01-01'),
+            new DateTime('2016-02-01')
+        );
+        $limit = 2;
+
+        $this->beConstructedThrough('createForPublicationIdWithTimeRangeAndLimit',[PublicationId::AP, $timeRange, $limit]);
```
Comment (filipgolonka) [inline spec/Snt/Capi/Repository/Article/FindParametersSpec.php]: as above

File: src/Snt/Capi/Repository/Article/FindParameters.php [hunk 3]
```diff
@@ -118,4 +169,23 @@ public function buildArticleIdsString($separator = self::DEFAULT_SEPARATOR)
...
+            $query .= sprintf('limit=%s', $this->limit);
+        }
+
+        if ($this->timeRange instanceof TimeRange) {
+            $query .= sprintf('&since=%s', $this->timeRange->getSince()->format(self::QUERY_DATETIME_FORMAT));
+            $query .= sprintf('&until=%s', $this->timeRange->getUntil()->format(self::QUERY_DATETIME_FORMAT));
+        }
+
+        return $query;
+    }
```
Comment (filipgolonka) [inline src/Snt/Capi/Repository/Article/FindParameters.php:190]: $query = [];
$query[] = 'foo';

return implode('&', $query);

**Comment by rafalkanski** [comment]
@filipgolonka review

**Comment by barkahl** [comment]
👍

File: src/Snt/Capi/Repository/Article/FindParameters.php [hunk 4]
```diff
@@ -118,4 +169,23 @@ public function buildArticleIdsString($separator = self::DEFAULT_SEPARATOR)
...
+        if (!is_null($this->limit)) {
+            $query[] = sprintf('limit=%s', $this->limit);
+        }
+
+        if ($this->timeRange instanceof TimeRangeParameter) {
+            $query[] = sprintf('since=%s', $this->timeRange->getSince()->format(self::QUERY_DATETIME_FORMAT));
+            $query[] = sprintf('until=%s', $this->timeRange->getUntil()->format(self::QUERY_DATETIME_FORMAT));
+        }
+
+        return implode('&', $query);
```
Comment (filipgolonka) [inline src/Snt/Capi/Repository/Article/FindParameters.php]: can be done even better using http://php.net/manual/en/function.http-build-query.php

:dancer: :dancer:

**Comment by rafalkanski** [comment]
@filipgolonka review

**Comment by filipgolonka** [comment]
:+1:

**Pull request merged**